ros_ws/src/dataset_generation/include/dataset_generation/filter_point_clouds.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <dirent.h>
+#include <iostream>
+#include <thread>
+#include <string>
+#include <sstream>
+#include <unordered_set>
+
+#include <pcl/ModelCoefficients.h>
+#include <pcl/PCLPointCloud2.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/common/io.h>
+#include <pcl/point_types.h>
+#include <pcl/common/common_headers.h>
+#include <pcl/filters/voxel_grid.h>
+
+#include "extract_point_cloud_objects.h"
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include "dataset_generation/json.hpp"
+
+using namespace std::literals::chrono_literals;
+
+namespace dataset_generation
+{
+
+
+class PCDFiltering
+{
+public:
+    PCDFiltering(
+        std::string in_folder_path,
+        std::string object_name,
+        std::string out_folder_path);
+
+    ~PCDFiltering();
+
+    void VoxelFilter(float leaf_size_x, float leaf_size_y, float leaf_size_z);
+private:
```

Comment:
It would be helpful to also add a `void apply_voxel_filter_and_visualize()` function, that visualizes the point cloud. Would be helpful with experimenting. You can check the visualization function in my extraction class for an example.